### ISSUE TITLE
Event loop hangs when ~100-1000 "stat" or "mkdir" promises are run in parallel

### ISSUE THREAD

**Issue by paul-thompson-helix** [denoland/deno#8335]
When approximately 100-1000 promises awaiting stat or mkdir async promises results are run in parallel the event loop hangs.

```
async function statHangs(numParallel: number) {
  const promises : Promise<any>[] = [];
  for(let i=0; i < numParallel; ++i) {
    promises.push(Deno.stat('./stat_test.ts'));
  }
  await Promise.all(promises);
}

// stat hangs when used with around 200 in parallel
statHangs(200);

async function mkdirHangs(numParallel: number) {
  const promises : Promise<any>[] = [];
  for(let i=0; i < numParallel; ++i) {
    promises.push(Deno.mkdir('/tmp/foo',{recursive:true}));
  }
  await Promise.all(promises);
}

// mkdir hangs eventually when too many are in parallel:
mkdirHangs(5000);

/// Adding an unrelated interval timer helps in borderline case ~200 in parallel.
/// The hang is resolved when the timer event fires.
/// Doesn't help for > 1000 in parallel.
async function statWithTimer(numParallel: number) {
  const aliveTimer = setInterval(()=>{
    console.log("event loop is alive"); // doesn't appear during the hang.
  }, 1000);

  const promises : Promise<any>[] = [];
  for(let i=0; i < numParallel; ++i) {
    promises.push(Deno.stat('./stat_test.ts'));
  }
  await Promise.all(promises);

  clearInterval(aliveTimer);
}

//statHangs(200);
//mkdirHangs(5000);
statWithTimer(200)
```

Notes:
* Original use case discovered this in https://deno.land/x/dnit@dnit-v1.6.1 task util where it needs to stat 100's or 1000's of files and I (lazily) implemented it with them all in parallel.  
* Test cases above repeat stat or mkdir on the same file but that is immaterial to the bug report.
* Obvious workaround is to have a task queue in javascript.
* Run with `strace` to watch syscalls to observe hanging behaviour.

tested with deno versions:
1.4.6
1.5.2

**Comment by Soremwar** [comment]
I can still reproduce this in latest canary

**Comment by caspervonb** [comment]
So this one isn't directly related to the resource table, I wouldn't expect any changes here from that.

Dug into it a bit, and this consistently hangs on epoll_wait.

**Comment by PaulThompson** [comment]
Appears to be resolved somewhere between 1.7.5 and 1.8.0

**Comment by caspervonb** [comment]
Confirmed, cannot reproduce.
